libzkbob-rs/src/client/state.rs

```diff
@@ -175,6 +175,40 @@ where
         .collect()
     }
 
+    pub fn get_account(&self, index: u64) -> Option<Account<P::Fr>> {
+        match self.txs.get(index) {
+            Some(Transaction::Account(acc)) => Some(acc),
+            _ => None,
+        }
+    }
+
+    pub fn get_previous_account(&self, index: u64) -> Option<(u64, Account<P::Fr>)> {
+        if index == 0 { return None }
+
+        let prev_acc_indexes = self.txs
+            .iter_slice(0..=(index-1))
+            .filter_map(|(idx, tx)| match tx {
+                Transaction::Account(_) => Some(idx),
+                _ => None,
+            })
+            .max();
+
+        match prev_acc_indexes {
+            Some(idx) => Some((idx, self.get_account(idx).unwrap())),
+            _ => None,
+        }
+    }
+
+    pub fn get_notes_in_range(&self, range: Range<u64>) -> Vec<(u64, Note<P::Fr>)> {
+        self.txs
+            .iter_slice(range.start..=(if range.end==0 { 0 } else { range.end-1 }))
```

Comment:
I guess we can replace it with `range.end.saturating_sub(1)`. Wdyt?

Reply:
Thanks, it looks more better

Fixed in https://github.com/zkBob/libzkbob-rs/pull/46/commits/c21ea801b890216200e3ed539fa6ea13e707f70c